libraries/AP_DAL/AP_DAL_RangeFinder.cpp

```diff
@@ -0,0 +1,104 @@
+#include "AP_DAL_RangeFinder.h"
+
+#include <AP_Logger/AP_Logger.h>
+
+#include <AP_RangeFinder/AP_RangeFinder_Backend.h>
+#include "AP_DAL.h"
+
+AP_DAL_RangeFinder::AP_DAL_RangeFinder()
+{
+    for (uint8_t i=0; i<ARRAY_SIZE(_RRNI); i++) {
+        _RRNI[i].instance = i;
+    }
+
+    for (uint8_t i=0; i<RANGEFINDER_MAX_INSTANCES; i++) {
+        // this avoids having to discard a const....
+        _backend[i] = new AP_DAL_RangeFinder_Backend(_RRNI[i]);
+    }
+}
+
+int16_t AP_DAL_RangeFinder::ground_clearance_cm_orient(enum Rotation orientation) const
+{
+#if !APM_BUILD_TYPE(APM_BUILD_AP_DAL_Standalone)
+    const auto *rangefinder = AP::rangefinder();
+
+    if (orientation != ROTATION_PITCH_270) {
+        // the EKF only asks for this from a specific orientation.  Thankfully.
+        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+        return rangefinder->ground_clearance_cm_orient(orientation);
+    }
+#endif
+
+    return _RRNH.ground_clearance_cm;
+}
+
+int16_t AP_DAL_RangeFinder::max_distance_cm_orient(enum Rotation orientation) const
+{
+#if !APM_BUILD_TYPE(APM_BUILD_AP_DAL_Standalone)
+    const auto *rangefinder = AP::rangefinder();
+
+    if (orientation != ROTATION_PITCH_270) {
+        // the EKF only asks for this from a specific orientation.  Thankfully.
+        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+        return rangefinder->ground_clearance_cm_orient(orientation);
+    }
+#endif
+
+    return _RRNH.max_distance_cm;
+}
+
+void AP_DAL_RangeFinder::start_frame()
+{
+    const auto *rangefinder = AP::rangefinder();
+    if (rangefinder == nullptr) {
+        return;
+    }
+
+    // EKF only asks for this *down*.
+    const log_RRNH old = _RRNH;
```

Comment:
```suggestion
    const log_RRNH old = _RRNH;

    // EKF only asks for this *down*.
```

Just reordering to match commnet with relevant code

Reply:
done